libsolidity/analysis/DocStringTagParser.cpp

```diff
@@ -162,6 +162,56 @@ bool DocStringTagParser::visit(ErrorDefinition const& _error)
 	return true;
 }
 
+bool DocStringTagParser::visit(InlineAssembly const& _assembly)
+{
+	if (!_assembly.documentation())
+		return true;
+	StructuredDocumentation documentation{-1, _assembly.location(), _assembly.documentation()};
+	ErrorList errors;
+	ErrorReporter errorReporter{errors};
+	auto docTags = DocStringParser{documentation, errorReporter}.parse();
+
+	if (!errors.empty())
+	{
+		SecondarySourceLocation ssl;
+		for (auto const& error: errors)
+			if (error->comment())
+				ssl.append(
+					*error->comment(),
+					error->sourceLocation() ? *error->sourceLocation() : _assembly.location()
+				);
+		m_errorReporter.warning(
```

Comment:
Is this a breaking change?

Reply:
The main point of doing all this locally with an artificual AST node with ID -1 and all that is to be able to translate the errors to warnings here, s.t. it's not breaking...

Reply:
ok!

Reply:
That's also implementation-wise the main thing I'm wondering about. We could also promote ``InlineAssembly`` to properly inherit ``StructurallyDocumented``, but then we need case distinctions in the "normal" natspec parsing logic to prevent it to become breaking on pain of stuff like this being proper errors.